assets/javascripts/modules/autoComplete.js

```diff
@@ -114,6 +114,7 @@ var closeSuggestionsAndFocus = function () {
  */
 var clearInput = function () {
   $autoCompleteInputElem.val('').focus();
+  $autoCompleteInputElem.data('suggestionValue', '');
```

Comment:
whatever you feel is best. Its more the same thing is being done in multiple places. Abstracting it into one fuction will tidy that up a bit